codelab/01_kpt-validate/README.md

```diff
@@ -0,0 +1,32 @@
+# Validate Your Configurations in skaffold Workflow
+
+This codelab introduces how to validate your app configurations before deploying them to the cluster. You are **not** required to know `kpt` or `skaffold` before starting the codelab. 
+
+After this session, you should be obtain more powerful tools to validate and manage your configurations.
+
+## What is kpt?
+
+Kpt is [an OSS tool](https://github.com/GoogleContainerTools/kpt) for Kubernetes packaging, which uses a standard format to bundle, publish, customize, update, and apply configuration manifests.
+
+## What kpt can help you in skaffold
+
+-  You can validate each of your config changes **declaratively**.
+-  You will get an hands-on off-the-shelf experience about the **GitOps** CI/CD workflow in skaffold.
+-  You **won't **encounter** version conflict** if the config hydration (a.k.a kustomize) mismatch with the deployment tool (e.g. kubectl). 
+-  You can prune your resources accurately with [a three-way merge strategy](https://kubectl.docs.kubernetes.io/pages/app_management/field_merge_semantics.html). 
+
+## What you'll learn
+
+-  How to add a validation to your config changes. 
+-  How to define validation rules in the form of a declarative pipeline.
+-  How to use the validation in kustomized resources. 
+-  How to reconcile your configuration changes with the live state
+
+
+You can try out this interactive codelab using [Cloud Shell](https://cloud.google.com/shell). 
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleContainerTools/skaffold&cloudshell_working_dir=codelab/01_kpt-validate&cloudshell_workspace=codelab/01_kpt-validate&cloudshell_tutorial=tutorial.md)
```

Comment:
FYI, this link will work once the PR is merged. To test out this PR, please use link 
https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/yuwenma/skaffold&cloudshell_working_dir=codelab/01_kpt-validate&cloudshell_workspace=codelab/01_kpt-validate&cloudshell_tutorial=tutorial.md

Reply:
tried this link but it still complains
```
Tutorial file /home/gaghosh/cloudshell_open/skaffold/codelab/01_kpt-validate/tutorial.md does not exist or is not a file
```

Reply:
Oh, that is because the link uses the "master" branch as default, but the tutorials.md  is in the "validate-1" branch.   

I've merged both validate-1 and deploy-2 branch to my master. They should work now. Do you mind trying again?